### PR TITLE
dev-python/traits: version bump

### PR DESCRIPTION
**Pull request by TheChymera**
Package-Manager: Portage-2.3.5, Repoman-2.3.2
And added myself to the maintainer list.

### CODE REVIEW

File: dev-python/traits/traits-4.6.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	fi
+}
+
+python_prepare_all() {
+	sed -i -e "s/'-O3'//g" setup.py || die
+	distutils-r1_python_prepare_all
+}
+
+python_compile() {
+	local -x CFLAGS="${CFLAGS} -fno-strict-aliasing"
```
Comment (mgorny) [inline dev-python/traits/traits-4.6.0.ebuild]: Is this necessary for Python 3? Please test whether gcc complains about strict aliasing problems when you comment it out.

**Comment by TheChymera** [comment]
@mgorny the tests pass. The CFLAG bit is indeed, not needed. The build log has nothing to say about aliasing.

**Comment by mgorny** [comment]
> The build log has nothing to say about aliasing.

Not even for Python 2? Hmm, I'm a bit confused now. @gentoo/python, any help? Did we solve the strict aliasing problem in Python 2.7 otherwise, or does it just not apply to this package at all?

**Comment by SoapGentoo** [comment]
I'm not aware of the latest python 2 fixing it, given that it would break the API. @TheChymera I would rather reintroduce it, unless you ensure that undefined behaviour provoked by type punning is not invoked.

**Comment by TheChymera** [comment]
> unless you ensure that undefined behaviour provoked by type punning is not invoked.

How could I do this?

**Comment by mgorny** [comment]
Ok, did some research and it seems that at some point Python just stopped passing `-Wall` to extensions. So to test it, please just:

    export CFLAGS='-O2 -Werror=strict-aliasing'
    emerge ... # (or ebuild ... compile )

This will cause the build to fail if strict aliasing warnings happen (and you don't have -fno-strict-aliasing).

Left notes in https://wiki.gentoo.org/wiki/Project:Python/Strict_aliasing

**Comment by TheChymera** [comment]
@mgorny thanks so much for writing that page to clarify things.

Indeed strict aliasing was broken for python 2, but works on python 3. The code you suggested fixes the issue.

File: dev-python/traits/traits-4.6.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+	fi
+}
+
+python_prepare_all() {
+	sed -i -e "s/'-O3'//g" setup.py || die
+	distutils-r1_python_prepare_all
+}
+
+python_compile() {
+	# include CXXFLAGS if C++ code is used
```
Comment (mgorny) [inline dev-python/traits/traits-4.6.0.ebuild]: This one was a comment for the reader, you weren't supposed to copy it ;-D.

**Comment by mgorny** [comment]
One more thing: since you're the maintainer now, could you look (without changing this pull request, possibly filing new) if the 2to3 call for testing is still necessary. If it is, could you look at requesting upstream to add those files to the 2to3 call in setup.py?

**Pull request closed without merging**